lxd/db_storage_volumes.go

```diff
@@ -0,0 +1,79 @@
+package main
+
+import (
+	"database/sql"
+
+	_ "github.com/mattn/go-sqlite3"
+)
+
+// Get config of a storage volume.
+func dbStorageVolumeConfigGet(db *sql.DB, volumeID int64) (map[string]string, error) {
+	var key, value string
+	query := "SELECT key, value FROM storage_volumes_config WHERE storage_volume_id=?"
+	inargs := []interface{}{volumeID}
+	outargs := []interface{}{key, value}
+
+	results, err := dbQueryScan(db, query, inargs, outargs)
+	if err != nil {
+		return nil, err
+	}
+
+	if len(results) == 0 {
+		/*
+		 * If we didn't get any rows here, let's check to make sure the
+		 * storage volume really exists; if it doesn't, let's send back
+		 * a 404.
+		 */
+		query := "SELECT id FROM storage_volumes WHERE id=?"
+		var r int
+		results, err := dbQueryScan(db, query, []interface{}{volumeID}, []interface{}{r})
+		if err != nil {
+			return nil, err
+		}
+
+		if len(results) == 0 {
+			return nil, NoSuchObjectError
+		}
+	}
```

Comment:
Same comment as for the storage pool. You need a volume id to call this function, to get that ID, you've retrieved it from its name or from a reference in the DB and so already know it's valid. So this code is unneeded and just causes more DB hits.